### PR TITLE
Add OSX support

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xhochy** [comment]
@conda-forge/mimalloc This is ready for review. I'll work afterwards on Windows support.

File: .ci_support/osx_.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+MACOSX_DEPLOYMENT_TARGET:
+- '10.9'
+c_compiler:
+- clang
+c_compiler_version:
+- '9'
```
Comment (elbaro) [inline .ci_support/osx_.yaml:6]: is this conda-forge's clang? then can we use 10?

Comment (xhochy) [reply .ci_support/osx_.yaml:6]: Not only in this recipe but you would need to update the pinnings at https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/ab8ca327dd58354b750e9d01a7daea287975c82e/recipe/conda_build_config.yaml#L12

Comment (xhochy) [reply .ci_support/osx_.yaml:6]: (and yes, this is conda-forge's clang)

**Pull request merged**